### ISSUE TITLE
Hang Xu

### ISSUE THREAD

**Issue by HangXXXu** [VIP-SMUR/24Fa-EnergyInBuildings-Com#6]
**Week 9.23 to 9.30**

**Tasks for This Week**
Proposed tasks for teams:
1.Rebuild an urban-scale model for energy simulation (Tech Square as a case)
2.Study about Google TensorFlow Smart Buildings Simulator 
3.Feature study for ML models

**Challenges & Solutions:**
1.Review existing paper related to ML-based energy simulation.
2.Find search gap for our project.
3.Identify ‘Building Orientation’ and ‘Footprint Shape’ as special features

**General Takeaway, Skills & Knowledge Acquired:**
For team management:
1.Customize and assign tasks to team members;
2.Make plan for the whole project.

**Team Meetings:**
Regular meeting on Tuesday and Friday

**Personal Reflection:**
Think more about how to organize the team and communicate efficiently.

**Plans for Next Week**
Need to study know to extract feature information for our use.

**Comment by HangXXXu** [comment]
## Weekly Notebook Entry — Week [04]
 
### Overview
- **Week Span:** `[9/30]` to `[10/7]`
 
### Tasks for This Week
Proposed future steps for our project as:
1.Extract geometry information from OSM file and merge them to existing energy demand dataset (we are here)
2.Based on building geometry, identify ‘uncommon’ feature for our own model, e.g footpring shape, shading
3.Develop algorithm to recognize and classify footprint shape 
4.Study about how to encode such non-numeric variable to regression model
5.Model training
6.Validate our surrogate model by comparison with parametric simulation

### Challenges & Solutions
Difficult to read and merge the datasets;
Key finding is that single OSM file can be merged with metadata via ‘bldg_id’;
Decided x-variable (columns) of the datasets.
 
### General Takeaway, Skills & Knowledge Acquired
For team management:
Customize and assign tasks to team members;
Make plan for the whole project.

For practical skills:
Python coding to automatically scraper data from website
 
### Team Meetings
Regular meeting on Tuesday and Friday
 
### Personal Reflection
Think more about how to organize the team and communicate efficiently.
 
### Plans for Next Week
Keep working on datasets.

**Comment by HangXXXu** [comment]
## Weekly Notebook Entry — Week [05]
 
### Overview
- **Week Span:** `[10/7]` to `[10/14]`
 
### Tasks for This Week
Proposed our research workflow as:
<div align="center">
    <img src="https://github.com/user-attachments/assets/ed0dc616-4fcb-4435-8d36-c6f2980af829" alt="workflow image">
</div>


### Challenges & Solutions
How to calculate numerical building features?
X1: Building Area = largest RoofCeiling surface
X2: Height = Z coordinate of the largest RoofCeiling surface
X3: Wall Area = Height * Perimeter of largest RoofCeiling surface
X4: WWR = Fixedwindow Area/(Wall Area-Fixedwindow Area) (Anu & Je)
X5: Orientation = OS:Building - North Axis  (Jo)
X6: Building Type = Office, Restaurant, Hotel …  
X7: Footprint Shape = ? (Benj & Jo)
Key is to rebuild building geometry in OSM files to calculate numerical building features.
 
### General Takeaway, Skills & Knowledge Acquired
For team management:
Customize and assign tasks to team members;
Make plan for the whole project.

For practical skills:
Python coding to multiply rebuild building geometry;
Python coding to automatically scraper data from website.

### Team Meetings
Regular meeting on Tuesday and Friday
 
### Personal Reflection
Even though most of feature is available, remain uncertainty about building footprint classification
 
### Plans for Next Week
Keep working on datasets;
OSM gives definition of building shape as ‘1,2,3...’, need to check it.

**Comment by HangXXXu** [comment]
## Weekly Notebook Entry — Week [06]
 
### Overview
- **Week Span:** `[10/14]` to `[10/21]`
 
### Tasks for This Week
Make plans and divide the work for GURC:
Anubha - Complete Training data
Joseph - Overall concept; Frontend / UI 
Jessica - Backend / Frontend (from Figma)
Hang - Train ML model
Han-Syun - Frontend (1st iteration on Figma)
Kiana - Overall concept / general help
Sharmista - Frontend (1st iteration on Figma)
Jiayi - Slide deck and storyline

### Challenges & Solutions
Design the prototype of web interface:
<div align="center">
    <img src="https://github.com/user-attachments/assets/c9f2e8a8-56f1-4d9b-b6c0-d4ec95f35ddc" alt="workflow image">
</div>

### General Takeaway, Skills & Knowledge Acquired
For team management:
Customize and assign tasks to team members.
Make plan for the whole project.

For practical skills:
Python coding to successfully extract all OSM file for Georgia region (21.5GB)

### Team Meetings
Regular meeting on Tuesday and Friday
 
### Personal Reflection
Every simulation geometry in OSM file is rectangle-based, even though they are labeled with different kind of shape information. I suppose that NREL simplify it for simulation process.
 
### Plans for Next Week
Complete the datasets.

**Comment by HangXXXu** [comment]
## Weekly Notebook Entry — Week [07]
 
### Overview
- **Week Span:** `[10/21]` to `[10/28]`
 
### Tasks for This Week
Proposed tasks for teams:
1.Default com building templete for comparison (Kiana, Jiayi) 
2.Front-end development (Anubha, Jessica)
3.Geometry previewer (Joseph, Benj, Anubha, Jessica, Jiayi, Kiana, Sharmista)
4.ML model training (Hang, Shivam)

### Challenges & Solutions
What is a comprehensive and professional workflow for ML training?
Self-learning, read paper and watch tutorial.

### General Takeaway, Skills & Knowledge Acquired
For team management:
Customize and assign tasks to team members.
Make plan for the whole project.

For practical skills:
Python coding to preprocess data for ML models
1.Read datasets
2.Clean the data (null and outliers)
3.Deal with highly correlated features to avoid multicollinearity
4.Encode non-numerical columns (one-hot, label, target)
5.Check the distribution of the numerical features and make logarithmic transformations if necessary

Data analysis result:
<div align="center">
    <img src="https://github.com/user-attachments/assets/d057aae8-751d-4dd5-ab10-0d34d1ed6fa8" alt="workflow image">
</div>

### Team Meetings
Regular meeting on Tuesday and Friday.
 
### Personal Reflection
Need to read more paper about ML model for energy prediction.
 
### Plans for Next Week
Train the ML models.

**Comment by HangXXXu** [comment]
## Weekly Notebook Entry — Week [08]
 
### Overview
- **Week Span:** `[10/28]` to `[11/4]`
 
### Tasks for This Week
Proposed ML- train workflow:
<div align="center">
    <img src="https://github.com/user-attachments/assets/95efc2c6-07e2-4c2f-98da-01c4ea8993b8" alt="workflow image">
</div>

Personally involved in:
1.Machine learning model training and comparison
2.Web application

### Challenges & Solutions
How to find the best ML for our datasets and implement it to web application?
After carefully paper review, I find that GB and tree-based model may perform better than MLR,  KNN, SVR, Random forest models from last semester

### General Takeaway, Skills & Knowledge Acquired
For practical skills:
Python coding (scikit-learn) to train ML models:
1.Based on datasets pre-processed last week, use computational software package to run MLR,SVR,KNN,XGB,RF,ET,LGB,GB,DT models for heating and cooling energy prediction
2.Select out GB as the best model, further use Bayesian optimizer to finetune model hyper-parameter for better R-square on training datasets.

Python coding to deploy ML model in Web interface:
1.Export encoder and model as (.pkl) and (.sav)
2.Modify deployment code in web app 

### Team Meetings
Regular meeting on Tuesday and Friday.
 
### Personal Reflection
Every point is covered for the conference.
 
### Plans for Next Week
Attend the conference.

**Comment by HangXXXu** [comment]
## Weekly Notebook Entry — Week [09]
 
### Overview
- **Week Span:** `[11/4]` to `[11/11]`
 
### Tasks for This Week
Enjoy the conference presentation;
Take a break.

**Comment by HangXXXu** [comment]
## Weekly Notebook Entry — Week [10]
 
### Overview
- **Week Span:** `[11/11]` to `[11/18]`
 
### Tasks for This Week
To complete last section of methodology proposed in Week 10.7 to 10.14, we have the following validation workflow:
<div align="center">
    <img src="https://github.com/user-attachments/assets/9a7c680b-4ede-4d31-8ca9-a11b3fd87c4d" alt="workflow image">
</div>

Proposed tasks for teams:
For architects (5)
1.run simulation 10 times (2 times for one person)
2.compare simulation results with ML prediction results
3.refer to simulation settings in OSM files (all footprint needs to be rectangle)

For computer scientist (3)
1.check the innovation, is there any existing ML-based energy prediction plugin for Rhino/Grasshopper (cove.tool)
2.is there any more wise way to build parametric model (Sampling → Modeling)
3.how to implement ML model in grasshopper

### Plans for Next Week
Prepare final documents for the VIP course.
